tests/integration_tests/test_grpc_only.py

```diff
@@ -58,33 +60,43 @@ def test_grpc_mode(custom_ethermint):
         "to": contract.address,
         "data": contract.encodeABI(fn_name="currentChainID"),
     }
-    api_port = ports.api_port(custom_ethermint.base_port(2))
+    api_port = ports.api_port(custom_ethermint.base_port(1))
     # in normal mode, grpc query works even if we don't pass chain_id explicitly
-    rsp = grpc_eth_call(api_port, msg)
-    print(rsp)
-    assert "code" not in rsp, str(rsp)
-    assert 9000 == int.from_bytes(base64.b64decode(rsp["ret"].encode()), "big")
-
+    success = False
+    for i in range(3):
+        rsp = grpc_eth_call(api_port, msg)
+        print(i, rsp)
+        assert "code" not in rsp, str(rsp)
+        ret = rsp["ret"]
+        valid = ret is not None
+        if valid and 9000 == int.from_bytes(base64.b64decode(ret.encode()), "big"):
+            success = True
+            break
+        time.sleep(1)
+    assert success
+    # wait 1 block before stop
+    for i in range(2):
+        wait_for_block(custom_ethermint.cosmos_cli(i), 1)
     supervisorctl(
-        custom_ethermint.base_dir / "../tasks.ini", "stop", "ethermint_9000-1-node2"
+        custom_ethermint.base_dir / "../tasks.ini", "stop", "ethermint_9000-1-node1"
     )
 
     # run grpc-only mode directly with existing chain state
-    with (custom_ethermint.base_dir / "node2.log").open("w") as logfile:
+    with (custom_ethermint.base_dir / "node1-new.log").open("w") as logfile:
```

Comment:
what is "node1-new"?

Reply:
to avoid previous overwrite log file after restart

Reply:
> to avoid overwrite previous node's log file after restart

please add that as a code comment